kedro/runner/runner.py

```diff
@@ -110,9 +116,10 @@ def run(
         free_outputs = pipeline.outputs() - (set(registered_ds) - memory_datasets)
 
         # Register the default dataset pattern with the catalog
```

Comment:
How does this work for the new Catalog?

Reply:
All the patterns are managed by `DataCatalogConfigResolver` which has `dataset_patterns`, `default_pattern` and `runtime_patterns` and stores them explicitly - https://github.com/kedro-org/kedro/blob/506470aa2386d7d7d5209cde9b53e09e47c4dcf3/kedro/io/catalog_config_resolver.py#L98

Previously we used shallow copy just to add `runtime_patterns` (line that you're pointing to).  Now we have a dedicated method for that - https://github.com/kedro-org/kedro/blob/506470aa2386d7d7d5209cde9b53e09e47c4dcf3/kedro/io/catalog_config_resolver.py#L246

We add runtime patterns to the `DataCatalogConfigResolver` at the level of the session where runner is already initialised, so runner does not depend on `DataCatalogConfigResolver`.
https://github.com/kedro-org/kedro/blob/506470aa2386d7d7d5209cde9b53e09e47c4dcf3/kedro/framework/session/session.py#L409